service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -213,12 +224,41 @@ private boolean dummySpecimenOrObservationExists(Specimen specimen, List<Observa
         return Optional.empty();
     }
 
+    private static String buildHumanName(Practitioner practitioner) {
+
+        if (practitioner.hasName()) {
+            var name = practitioner.getNameFirstRep();
+            var nameBuilder = new StringBuilder();
+
+            if (name.hasPrefix()) {
+                nameBuilder.append(name.getPrefix());
+                nameBuilder.append(StringUtils.SPACE);
+            }
+
+            if (name.hasGiven()) {
+                nameBuilder.append(name.getGivenAsSingleString());
+                nameBuilder.append(StringUtils.SPACE);
+            }
+
+            if (name.hasFamily()) {
+                nameBuilder.append(name.getFamily());
+            }
+
+            if (StringUtils.isNotBlank(nameBuilder.toString())) {
```

Comment:
Is there any need for this condition? In addition you are building a string here twice.